lib/http/HttpClient_Apple.mm

```diff
@@ -29,6 +29,9 @@
     return std::string("RESP-") + std::to_string(seq.fetch_add(1));
 }
 
+static dispatch_once_t once;
+static NSURLSession* m_session = nil;
```

Comment:
Can't these static's be member variables of HttpRequestApple ?

Reply:
Yes I have tried to do the same. The HttpRequestApple class is defined in .mm file, I was getting a compilation error if I defined the static objects in .mm. A workaround for this is to define the HttpRequestApple class inside .hh header and define the static variables in the header itself. However that exposes the whole HttpRequestApple class to files which import the header. Hence I have created static objects outside of the class.

Reply:
Can't we make it split it in the cc/h file, and ensure the source file is linked when building for iOS?

Reply:
> Can't these static's be member variables of HttpRequestApple ?

Indeed, this static variable could definitively be moved to the function itself as a local static.

Reply:
@samirnen  - Do you think this is doable?

Reply:
I tried moving the variable to the function as a local static. But then the memory is leaking same as before. So didn't go ahead with that approach

Reply:
I have a create a new source file for the HTTPRequestApple class and have defined the static variables there